mooncake-store/src/master_service.cpp

```diff
@@ -378,21 +370,33 @@ ErrorCode MasterService::Remove(const std::string& key) {
 long MasterService::RemoveAll() {
     long removed_count = 0;
     uint64_t total_freed_size = 0;
+    auto now = std::chrono::steady_clock::now();
+
     for (auto& shard : metadata_shards_) {
         std::unique_lock lock(shard.mutex);
-        std::size_t object_count = shard.metadata.size();
-        if (object_count == 0)
+        if (shard.metadata.empty()) {
             continue;
-        for (const auto& [key, metadata] : shard.metadata) {
-            // Calculate the total freed size of the object
-            total_freed_size += metadata.size * metadata.replicas.size();
         }
-        removed_count += object_count;
-        // Reset related metrics on successful removed
+
+        // Only remove objects with expired leases
+        auto it = shard.metadata.begin();
+        while (it != shard.metadata.end()) {
+            if (it->second.lease_timeout <= now) {
```

Comment:
We could add an is_lease_expired() function that gets the current time and checks if the lease is expired internally?

Reply:
Sounds good